crates/sgx_validation/src/config.rs

```diff
@@ -0,0 +1,146 @@
+    /// This is the list of public keys which have signed the Enarx binary.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub enarx_signer: Option<Vec<Vec<u8>>>,
+
+    /// Prohibited values for `mrsigner` in the report body.
+    /// This is the list of public keys which have signed the Enarx binary which are disallowed.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub enarx_signer_blacklist: Option<Vec<Vec<u8>>>,
+
+    /// Values for `mrenclave` in the report body.
+    /// This is the hash of the Enclave environment after the Enarx binary is loaded
+    /// but before any workload is loaded, so this is a hash of the Enarx binary
+    /// in memory.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub enarx_hash: Option<Vec<Vec<u8>>>,
+
+    /// Prohibited values for `mrenclave` in the report body.
+    /// This is the hash of the Enclave environment after the Enarx binary is loaded
+    /// but before any workload is loaded, so this is a hash of the Enarx binary
+    /// in memory.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub enarx_hash_blacklist: Option<Vec<Vec<u8>>>,
+
+    /// Values for `features`.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_features")]
+    pub features: Option<u64>,
+
+    /// Minimum value allowed for `cpusvn`.
+    pub cpu_svn: Option<Vec<u8>>,
+
+    /// Minimum value for `isv_svn`.
+    pub enclave_security_version: Option<u16>,
+
+    /// Value for `isv_prodid`, do not allow versions below this.
+    pub enclave_product_id: Option<u16>,
+}
+
+fn from_hex<'de, D>(deserializer: D) -> Result<Option<Vec<Vec<u8>>>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: Vec<&str> = Deserialize::deserialize(deserializer)?;
+
+    let mut outer_vec = Vec::new();
+    for hash_string in s {
+        outer_vec.push(hex::decode(hash_string).map_err(|_| Error::custom("invalid hex"))?);
+    }
+
+    Ok(Some(outer_vec))
+}
+
+fn from_features<'de, D>(deserializer: D) -> Result<Option<u64>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: &str = Deserialize::deserialize(deserializer)?;
+
+    let mut flags = Features::empty();
+
+    flags |= Features::INIT; // Must be set
+    flags |= Features::MODE64BIT; // Isn't everything 64-bit?
+
+    for flag in s.to_string().split('|') {
+        match flag.trim() {
+            "CET" => {
+                flags |= Features::CET;
+            }
+            "Debug" => {
+                flags |= Features::DEBUG;
+            }
+            "Eint_Key" => {
+                flags |= Features::EINIT_KEY;
+            }
+            "KSS" => {
+                flags |= Features::KSS;
+            }
+            "Provisioning_Key" => {
+                flags |= Features::PROVISIONING_KEY;
+            }
+            _ => return Err(D::Error::custom(format!("unknown flag '{}'", flag))),
+        }
+    }
+
+    Ok(Some(flags.bits()))
+}
```

Comment:
I thought about pushing this upwards and having the serde derives added to the SGX crate. Otherwise, I don't see a way to avoid something like this. I also couldn't figure out how to make use of `bitflags` use of serde.

Reply:
Changed so that Serde works with the strings via an Enum, so no strings are directly compared here. See if that works for you.

Reply:
Fixed